Fortnox.NET/WebSockets/FortnoxWebSocketClient.cs

```diff
@@ -107,161 +92,89 @@ private async Task<T> SendAndRecieveOnce<T>(string message)
                 }
             }
 
-            var deserializedResult = JsonConvert.DeserializeObject<T>(resultString, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
-            return deserializedResult;
-        }
-
-        /// <summary>
-        /// Helper function that allows the caller to specify a callback function that accepts the underlying <see cref="System.Net.WebSockets.ClientWebSocket"/>.
-        /// </summary>
-        /// <param name="callback">An asynchronous callback function that accepts a <see cref="System.Net.WebSockets.ClientWebSocket"/> in its parameters.</param>
-        /// <returns>An empty <see cref="Task"/></returns>
-        public async Task Listen(Func<ClientWebSocket, Task> callback)
-        {
-            await callback.Invoke(this._webSocket);
-        }
-
-        /// <summary>
-        /// Initiates the connection to the underlying <see cref="System.Net.WebSockets.ClientWebSocket"/> as well as subscribes to all the specified topics as an asynchronous operation.
-        /// </summary>
-        /// <returns>The current instance of <see cref="FortnoxWebSocketClient"/> as an asynchronous operation.</returns>
-        public async Task<FortnoxWebSocketClient> Connect()
-        {
-            // TODO(Oskar): Should we keep anonymous objects or create something concrete?
-            await _webSocket.ConnectAsync(_fortnoxWebSocketURI, CancellationToken.None);
-
-            var AddTenantsObject = new { command = WebSocketCommands.AddTenants, clientSecret = _clientSecret, accessTokens = new List<string>(_accessTokens) };
-            var AddTenantsResult = await SendAndRecieveOnce<WebSocketCommandResponse>(JsonConvert.SerializeObject(AddTenantsObject));
-
-            if (AddTenantsResult.Result.Equals("error"))
-            {
-                throw new WebSocketException("Unable to add tenants.");
-            }
-
-            if (_topics.Count <= 0)
+            var deserializedResult = JsonConvert.DeserializeObject<WebSocketResponse>(resultString, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
+            if (deserializedResult.TenantId.HasValue)
```

Comment:
prefer  `== null` over `HasValue` unless there is a specific reason not to

Reply:
This is of type long? which cannot be null.

Reply:
nullable long AKA `Nullable<long>` aka `long?` should def. be able to have the value null.

Reply:
Resolved